Source/WebKit/NetworkProcess/Classifier/ResourceLoadStatisticsStore.cpp

```diff
@@ -2160,7 +2163,7 @@ void ResourceLoadStatisticsStore::dumpResourceLoadStatistics(CompletionHandler<v
 {
     ASSERT(!RunLoop::isMain());
     if (m_dataRecordsBeingRemoved) {
-        m_dataRecordRemovalCompletionHandlers.append([this, completionHandler = WTFMove(completionHandler)]() mutable {
+        m_dataRecordRemovalCompletionHandlers.append([this, protectedThis = Ref { *this }, completionHandler = WTFMove(completionHandler)]() mutable {
```

Comment:
This looks like a ref-cycle to me and thus leaking. This needs to use `weakThis`.